### PR TITLE
Enhance the pattern search

### PR DESCRIPTION
**Pull request by dzhengfy**
Extend the search content type to list and str. Originally only str is supported.

### CODE REVIEW

File: virttest/utils_test/libvirt.py [hunk 1]
```diff
@@ -3671,14 +3671,19 @@ def check_cmd_output(cmd, content, err_ignore=False, session=None):
     """
     if cmd is None:
         raise exceptions.TestFail("cmd can not be None")
+    logging.debug("check_cmd_output()")
+    if not isinstance(content, str) and not isinstance(content, list):
```
Comment (chunfuwen) [inline virttest/utils_test/libvirt.py:3675]: The native string (str) – text in Python 3, bytes in Python 2, it may need be cared about

**Comment by chunfuwen** [comment]
Any available result?

**Comment by dzhengfy** [comment]
> Any available result?

The change impacts the case:
```
# avocado run --vt-type libvirt --vt-machine-type q35 virtual_devices.filesystem_device.positive_test.socket_file_checking.fs_test.xattr_on.flock_on.lock_posix_on.cache_mode_none.multiple_fs.one_guest
JOB ID     : 0f1f8fd01b3f60f07a16f993c73a239fc60e30aa
JOB LOG    : /root/avocado/job-results/job-2020-07-07T14.37-0f1f8fd/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virtual_devices.filesystem_device.positive_test.socket_file_checking.fs_test.xattr_on.flock_on.lock_posix_on.cache_mode_none.multiple_fs.one_guest: PASS (84.00 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 86.33 s
```
And some pci cases also use this PR. See https://github.com/autotest/tp-libvirt/pull/2601#issuecomment-654878103 and https://github.com/autotest/tp-libvirt/pull/2601#issuecomment-654877808

File: virttest/utils_test/libvirt.py (hunk 1 above)
Comment (dzhengfy) [reply virttest/utils_test/libvirt.py]: To avoid this unnecessary confusion, I remove this part. Please double check.

**Pull request merged**